src/gflownet/algo/trajectory_balance.py

```diff
@@ -363,7 +363,7 @@ def compute_batch_losses(
             traj_losses = self.subtb_loss_fast(log_p_F, log_p_B, per_graph_out[:, 0], clip_log_R, batch.traj_lens)
             # The position of the first graph of each trajectory
             first_graph_idx = torch.zeros_like(batch.traj_lens)
-            first_graph_idx = torch.cumsum(batch.traj_lens[:-1], 0, out=first_graph_idx[1:])
+            torch.cumsum(batch.traj_lens[:-1], 0, out=first_graph_idx[1:])
```

Comment:
is this a typo?

Reply:
No actually the previous version was an error.
```
# I was doing
b = [1,2,3]
a = [0,0,0]
a = cumsum(b[:-1]) 
# now = a = [1, 3] because that's what cumsum returns
# but doing
a = [0,0,0]
cumsum(b[:-1], out=a[:1]) 
# is the same as
a[:1] = cumsum(b[:-1])
# now a is [0, 1, 3]
```

Reply:
oh, i see